framework/src/main/java/org/tron/core/services/http/ListProposalsServlet.java

```diff
@@ -8,29 +9,43 @@
 import org.tron.api.GrpcAPI.ProposalList;
 import org.tron.core.Wallet;
 
-
 @Component
 @Slf4j(topic = "API")
 public class ListProposalsServlet extends RateLimiterServlet {
 
   @Autowired
   private Wallet wallet;
 
+  @Override
   protected void doGet(HttpServletRequest request, HttpServletResponse response) {
     try {
       boolean visible = Util.getVisible(request);
-      ProposalList reply = wallet.getProposalList();
-      if (reply != null) {
-        response.getWriter().println(JsonFormat.printToString(reply, visible));
-      } else {
-        response.getWriter().println("{}");
-      }
+      response(response, visible);
     } catch (Exception e) {
       Util.processError(e, response);
     }
   }
 
+  @Override
   protected void doPost(HttpServletRequest request, HttpServletResponse response) {
-    doGet(request, response);
+    try {
+      PostParams params = PostParams.getPostParams(request);
+      boolean visible = Util.getVisible(request);
```

Comment:
Why get it again here?

Reply:
To be compatible with the post method for obtaining parameters, when the visible obtained in the URL is false, it is then obtained from the post method.